src/compiler/utilities.ts

```diff
@@ -3298,6 +3297,10 @@ namespace ts {
         return false;
     }
 
+    export function isJSDocTypeReference(node: TypeReferenceType): node is TypeReferenceNode {

```

Comment:
So if you have `let o: ...object;` in a JS file, you will see `object` and climb up the tree to determine whether it's in a JSDoc comment? Since that's a grammar error anyway, I don't think it's a big deal if we give the `object` there TS semantics instead of JSDoc semantics.